gdx/src/com/badlogic/gdx/maps/tiled/BaseTmjMapLoader.java

```diff
@@ -0,0 +1,746 @@
+			break;
+		case "objectgroup":
+			loadObjectGroup(map, parentLayers, element);
+			break;
+		case "imagelayer":
+			loadImageLayer(map, parentLayers, element, tmjFile, imageResolver);
+			break;
+		}
+	}
+
+	protected void loadLayerGroup (TiledMap map, MapLayers parentLayers, JsonValue element, FileHandle tmjFile,
+		ImageResolver imageResolver) {
+		if (element.getString("type", "").equals("group")) {
+			MapGroupLayer groupLayer = new MapGroupLayer();
+			loadBasicLayerInfo(groupLayer, element);
+
+			JsonValue properties = element.get("properties");
+			if (properties != null) {
+				loadProperties(groupLayer.getProperties(), properties);
+			}
+			for (JsonValue child : element) {
+				loadLayer(map, groupLayer.getLayers(), child, tmjFile, imageResolver);
+			}
+
+			for (MapLayer layer : groupLayer.getLayers()) {
+				layer.setParent(groupLayer);
+			}
+
+			parentLayers.add(groupLayer);
+		}
+	}
+
+	protected void loadTileLayer (TiledMap map, MapLayers parentLayers, JsonValue element) {
+
+		if (element.getString("type", "").equals("tilelayer")) {
+			int width = element.getInt("width", 0);
+			int height = element.getInt("height", 0);
+			int tileWidth = map.getProperties().get("tilewidth", Integer.class);
+			int tileHeight = map.getProperties().get("tileheight", Integer.class);
+			TiledMapTileLayer layer = new TiledMapTileLayer(width, height, tileWidth, tileHeight);
+
+			loadBasicLayerInfo(layer, element);
+
+			int[] ids = getTileIds(element, width, height);
+			TiledMapTileSets tileSets = map.getTileSets();
+			for (int y = 0; y < height; y++) {
+				for (int x = 0; x < width; x++) {
+					int id = ids[y * width + x];
+					boolean flipHorizontally = ((id & FLAG_FLIP_HORIZONTALLY) != 0);
+					boolean flipVertically = ((id & FLAG_FLIP_VERTICALLY) != 0);
+					boolean flipDiagonally = ((id & FLAG_FLIP_DIAGONALLY) != 0);
+
+					TiledMapTile tile = tileSets.getTile(id & ~MASK_CLEAR);
+					if (tile != null) {
+						Cell cell = createTileLayerCell(flipHorizontally, flipVertically, flipDiagonally);
+						cell.setTile(tile);
+						layer.setCell(x, flipY ? height - 1 - y : y, cell);
+					}
+				}
+			}
+			JsonValue properties = element.get("properties");
+			if (properties != null) {
+				loadProperties(layer.getProperties(), properties);
+			}
+			parentLayers.add(layer);
+		}
+	}
+
+	protected void loadObjectGroup (TiledMap map, MapLayers parentLayers, JsonValue element) {
+		if (element.getString("type", "").equals("objectgroup")) {
+			MapLayer layer = new MapLayer();
+			loadBasicLayerInfo(layer, element);
+			JsonValue properties = element.get("properties");
+			if (properties != null) {
+				loadProperties(layer.getProperties(), properties);
+			}
+
+			for (JsonValue objectElement : element.get("objects")) {
+				loadObject(map, layer, objectElement);
+			}
+
+			parentLayers.add(layer);
+		}
+	}
+
+	protected void loadImageLayer (TiledMap map, MapLayers parentLayers, JsonValue element, FileHandle tmjFile,
```

Comment:
I have gone ahead and updated the loadImageLayer method to fix all the issues.
Firstly you'll notice I've added the repeatX and repeatY code, needed to match the most recent Master version of BaseTmxMapLoader.

But more importantly there was an error loading in image layers. You were looking for the 'source' key. But that property/attribute does not exist in the json files. You can get the image file source right from the image key.

```
protected void loadImageLayer (TiledMap map, MapLayers parentLayers, JsonValue element, FileHandle tmjFile,
		 ImageResolver imageResolver) {
		  if (element.getString("type", "").equals("imagelayer")) {
				float x = element.getFloat("offsetx", 0);
				float y = element.getFloat("offsety", 0);
				if (flipY) y = mapHeightInPixels - y;

				String imageSrc = element.getString("image","");

				boolean repeatX = element.getInt("repeatx", 0) == 1;
				boolean repeatY = element.getInt("repeaty", 0) == 1;

				TextureRegion texture = null;

				if (!imageSrc.isEmpty()) {
					 FileHandle handle = getRelativeFileHandle(tmjFile, imageSrc);
					 texture = imageResolver.getImage(handle.path());
					 y -= texture.getRegionHeight();
				}

				TiledMapImageLayer layer = new TiledMapImageLayer(texture, x, y, repeatX, repeatY);

				loadBasicLayerInfo(layer, element);

				JsonValue properties = element.get("properties");
				if (properties != null) {
					 loadProperties(layer.getProperties(), properties);
				}

				parentLayers.add(layer);
		  }
	 }
```